code/datums/elements/food/food_trash.dm

```diff
@@ -48,6 +56,8 @@
 		food_holding_mob.dropItemToGround(edible_object)
 		food_holding_mob.put_in_hands(trash_item)
 
+	Detach(source)
```

Comment:
Elements already call this when the item they are attached to is deleted, is source supposed to be deleted here? If so, shouldnt we do that instead?

Reply:
No, the source is not deleted when a food is used as an ingredient in a `customizable_reagent_holder`. Putting a food item in a custom pizza/sushi/whatever does not delete the food, instead it stores it in `var/list/obj/item/ingredients` of the component, and uses this list in further operations.
 
This is the only instance of behaviour where the food item is not deleted after or during this element being called.

Reply:
then I think you should make ``COMSIG_ITEM_USED_AS_INGREDIENT`` register to a different proc that will generate trash, then delete itself (the element) instead

Or maybe the element should always be deleting itself on this signal, for future edge-cases where items aren't deleted after generating trash?

Reply:
Yeah, I wanted to future proof the element, this is why I did not do a separate proc, and because the trash generation itself is otherwise identical.

I also considered actually deleting the source food item, and just add type paths in the ingredient list, however that would have made us lose dynamic data, such as returning the new name of the food (either due to a chef renaming it, or it being an [Assistantnamehere] Burger) for the descriptive list, or dynamically changing icon states (later one is used to generate the filling colour).

Reply:
Would the following split and autodoc comments be an acceptable alternative?

```
/// Call this in most cases where the source item is deleted
/datum/element/food_trash/proc/async_generate_trash(datum/source)
	generic_generate_trash()

/// Call this variant if the source item is not deleted on trash generation, for example, storing a reference to it in custom foods
/datum/element/food_trash/proc/async_generate_trash_and_detach(datum/source)
	generic_generate_trash()
	Detach(source)

/datum/element/food_trash/proc/generic_generate_trash(datum/source)
/* the generic trash generation proc is moved here*/
```

Reply:
Actually, this is a component not element, my bad this fix is fine.